apps/wallet/src/ui/app/components/sui-apps/SuiApp.tsx

```diff
@@ -19,80 +18,123 @@ export type DAppEntry = {
+type CardViewProp = {
+	name: string;
+	link: string;
+	icon?: string;
+};
+
+function CardView({ name, link, icon }: CardViewProp) {
 	const appUrl = getDAppUrl(link);
 	const originLabel = appUrl.hostname;
+	return (
+		<div className="bg-white group flex flex-col p-3.75 box-border w-full rounded-2xl border border-gray-45 border-solid h-32 hover:bg-sui/10 hover:border-sui/30">
+			<div className="flex mb-1">
+				<ImageIcon src={icon || null} label={name} fallback={name} size="lg" circle />
+			</div>
 
-	const AppDetails = (
-		<div className={cl(st.suiApp, st[displayType])}>
-			<div className={st.icon}>
-				{icon ? (
-					<img src={icon} className={st.icon} alt={name} />
-				) : (
-					<div className={st.defaultImg}></div>
-				)}
+			<div className="flex flex-col gap-1 justify-start item-start">
+				<div className="line-clamp-2 break-all">
+					<Text variant="body" weight="semibold" color="gray-90">
+						{name}
+					</Text>
+				</div>
+				<Text variant="bodySmall" weight="medium" color="steel" truncate>
+					{originLabel}
+				</Text>
 			</div>
-			<div className={st.info}>
-				<div className={st.title}>{name} </div>
-				{displayType === 'full' && <div className={st.description}>{description}</div>}
+		</div>
+	);
+}
 
-				{displayType === 'card' && <div className={st.link}>{originLabel}</div>}
+type ListViewProp = {
+	name: string;
+	icon?: string;
+	description: string;
+	tags?: string[];
+};
 
-				{displayType === 'full' && tags?.length && (
-					<div className={st.tags}>
+function ListView({ name, icon, description, tags }: ListViewProp) {
+	return (
+		<div className="bg-white group flex py-3.5 px-1.25 gap-3 item-center box-border rounded hover:bg-sui/10">
+			<ImageIcon src={icon || null} label={name} fallback={name} size="xl" circle />
+			<div className="flex flex-col gap-1 justify-center">
+				<Text variant="body" weight="semibold" color="sui-dark">
+					{name}
+				</Text>
+				<Text variant="subtitle" weight="normal" color="steel-darker">
+					{description}
+				</Text>
+				{tags?.length && (
+					<div className="flex flex-wrap gap-1">
 						{tags?.map((tag) => (
-							<div className={st.tag} key={tag}>
-								{tag}
+							<div
+								className="flex item-center justify-center px-1.5 py-0.5 border border-solid border-steel rounded"
+								key={tag}
+							>
+								<Text variant="captionSmall" weight="medium" color="steel-dark">
+									{tag}
+								</Text>
 							</div>
 						))}
 					</div>
 				)}
 			</div>
 		</div>
 	);
+}
+
+export interface SuiAppProps {
+	name: string;
+	description: string;
+	link: string;
+	icon: string;
+	tags: string[];
+	permissionID?: string;
+	displayType: DisplayType;
+}
+
+export function SuiApp({
+	name,
+	description,
+	link,
+	icon,
+	tags,
+	permissionID,
+	displayType,
+}: SuiAppProps) {
+	const [showDisconnectApp, setShowDisconnectApp] = useState(false);
+	const appUrl = getDAppUrl(link);
+
+	if (permissionID && showDisconnectApp) {
```

Comment:
We need a way to go to the DisconnectApp view when user is in connected apps page